analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClearTextProtocolsAreSensitive.cs

```diff
@@ -0,0 +1,196 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Mail;
+
+namespace Tests.Diagnostics
+{
+    class ClearTextProtocols
+    {
+        private const string a = "http://example.com"; // Noncompliant {{Using http protocol is insecure. Use https instead.}}
+        private const string b = "https://example.com";
+        private const string c = "http://localhost";
+        private const string d = "http://127.0.0.1";
+        private const string e = "http://::1";
+
+        private const string f = @"telnet://anonymous@example.com"; // Noncompliant {{Using telnet protocol is insecure. Use ssh instead.}}
+        private const string g = @"ssh://anonymous@example.com";
+        private const string h = @"telnet://anonymous@localhost";
+        private const string i = "telnet://anonymous@127.0.0.1";
+        private const string j = "telnet://anonymous@::1";
+
+        private readonly string k = $"ftp://anonymous@example.com"; // Noncompliant {{Using ftp protocol is insecure. Use sftp, scp or ftps instead.}}
+        private readonly string l = $"sftp://anonymous@example.com";
+        private readonly string m = $"ftp://anonymous@localhost";
+        private readonly string n = $"ftp://anonymous@127.0.0.1";
+        private readonly string o = $"ftp://anonymous@::1";
+
+        public void Method(string part, string user, string domain, string ftp)
+        {
+            var a = "http://example.com"; // Noncompliant
+            var b = "https://example.com";
+            var c = "http://localservername"; // Noncompliant, it's not a "localhost"
+            var d = "http://localhosting";
+            var e = "See http://www.example.com for more information";
+            var httpProtocolScheme = "http://"; // It's compliant when standalone
+
+            var f = $"ftp://anonymous@example.com"; // Noncompliant
+            var g = $"ftp://{part}@example.com"; // Noncompliant
+            var h = "ssh://anonymous@example.com";
+            var i = "sftp://anonymous@example.com";
+            var ftpProtocolScheme = "ftp://"; // It's compliant when standalone
+            var j = "ftp://" + user + "@example.com"; // Compliant - FN (protocol is compliant when standallone, check previous case)
+            var k = "ftp://anonymous@" + domain;// Noncompliant
+            var l = $"ftp://anonymous@{domain}"; // Noncompliant
+            var m = $"{ftp}://anonymous@example.com"; // Compliant - FN
```

Comment:
Why FN? It's true negative for me, as the ftp variable might contain anything
```suggestion
            var m = $"{ftp}://anonymous@example.com"; // Compliant
```